crypto/dh/dh_group_params.c

```diff
@@ -16,18 +18,34 @@
 
 static DH *dh_param_init(const BIGNUM *p, int32_t nbits)
 {
+    BIGNUM *q = NULL;
     DH *dh = DH_new();
+
     if (dh == NULL)
         return NULL;
+
+    q = BN_dup(p);
+    /* Set q = (p - 1) / 2 (p is known to be odd so just shift right ) */
+    if (q == NULL || !BN_rshift1(q, q)) {
+        BN_free(q);
+        DH_free(dh);
+        return NULL;
+    }
     dh->p = (BIGNUM *)p;
+    dh->q = (BIGNUM *)q;
     dh->g = (BIGNUM *)&_bignum_const_2;
-    dh->length = nbits;
+    dh->length = nbits; /* Private key length = 2 * s */
```

Comment:
What does "s" refer to here?

Reply:
It is listed in the description in DH_new_by_nid()

Reply:
Ah - right. No where does it refer to this as "s" though. Perhaps change this to "2 * max_target_security_strength" to be consistent with the description in DH_new_by_nid()?

Reply:
's' would have come out of the docs that I was referring to - will change the name..